cmd/templ/generatecmd/eventhandler.go

```diff
@@ -221,7 +222,12 @@ func (h *FSEventHandler) generate(ctx context.Context, fileName string) (goUpdat
 
 	formattedGoCode, err := format.Source(b.Bytes())
 	if err != nil {
-		return false, false, nil, fmt.Errorf("%s source formatting error: %w", fileName, err)
+		mapped, ok := mapFormatterError(err, sourceMap)
```

Comment:
As per the other comment, I think it would be better to look at the type of the error, and maybe even to rewrite the error that comes back.

A small test around that behaviour would be useful too, so we don't accidentally lose the behaviour in future work.

Reply:
Agreed, I'm stepping away for a bit and wanted to commit what I've got so far but I will add tests soon.